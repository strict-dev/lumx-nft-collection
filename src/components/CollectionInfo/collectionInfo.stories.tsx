import { Story, Meta } from '@storybook/react'
import CollectionInfo from '.'
export default {
  title: 'CollectionInfo',
  component: CollectionInfo
} as Meta

export const Default: Story = () => <CollectionInfo />
