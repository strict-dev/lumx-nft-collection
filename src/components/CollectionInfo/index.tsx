import UserPhoto from '@public/assets/bored-img.svg'
import MonitorIcon from '@public/assets/icn-monitor-list.svg'
import OpenSeaIcon from '@public/assets/icn-opensea-list.svg'
import EtherScanIcon from '@public/assets/icon-etherscan-list.svg'
import CopyIcon from '@public/assets/icn-copy.svg'
import TrendingUpIcon from '@public/assets/trending-up.svg'

import Image from 'next/image'
import { data } from 'utils/mock'
import CollectionInfoGridItem from 'components/CollectionInfoGridItem'
import CardWrapper from 'components/CardWrapper'
import Container from 'components/Container'

const CollectionInfo = () => {
  return (
    <Container>
      <CardWrapper className="mb-14">
        <div className="relative flex items-center justify-between gap-2 pb-4">
          <div className="flex items-center justify-between gap-2">
            <Image src={UserPhoto} alt="User" width={32} height={32} />
            <div className="flex flex-col justify-between">
              <h1 className="text-base font-bold">{data.collectionName}</h1>

              <div className="flex">
                <p className="text-sm font-normal truncate max-w-[140px] text-neutral-600">
                  {data.token}
                </p>
                <button>
                  <Image src={CopyIcon} alt="User" width={16} height={16} />
                </button>
              </div>
            </div>
          </div>

          <div className="flex gap-2 ">
            <Image
              src={MonitorIcon}
              alt="Monitor Icon"
              width={32}
              height={32}
            />
            <Image
              src={OpenSeaIcon}
              alt="OpenSea Icon"
              width={32}
              height={32}
            />
            <Image
              src={EtherScanIcon}
              alt="EtherScan Icon"
              width={32}
              height={32}
            />
          </div>

          <div className="absolute bottom-0 left-0 w-full border-b border-b-neutral-700" />
        </div>

        <div className="grid grid-flow-row grid-cols-3 grid-rows-2 gap-2 pt-4">
          <CollectionInfoGridItem title={'lifetime'} data={data.lifetime} />

          <CollectionInfoGridItem title={'items'} data={data.items} />

          <CollectionInfoGridItem title={'floor'} data={data.floor} />

          <CollectionInfoGridItem title={'royalties'} data={data.royalties} />

          <CollectionInfoGridItem title={'vol24h'} data={data.floor} type="vol">
            <div className="flex items-center justify-center gap-1">
              <Image src={TrendingUpIcon} alt="Icon" width={16} height={16} />
              <span className="font-bold text-support-success">
                {data.vol24h.performance}
              </span>
            </div>
          </CollectionInfoGridItem>

          <CollectionInfoGridItem title={'vol7d'} data={data.floor} type="vol">
            <div className="flex items-center justify-center gap-1">
              <Image src={TrendingUpIcon} alt="Icon" width={16} height={16} />
              <span className="font-bold text-support-error">
                {data.vol7d.performance}
              </span>
            </div>
          </CollectionInfoGridItem>
        </div>
      </CardWrapper>
    </Container>
  )
}

export default CollectionInfo
