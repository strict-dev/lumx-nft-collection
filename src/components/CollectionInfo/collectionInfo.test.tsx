import { render, screen } from '@testing-library/react'
import CollectionInfo from '.'

describe('<CollectionInfo />', () => {
  it('should render the heading', () => {
    const { container } = render(<CollectionInfo />)

    expect(screen.getByRole('heading', { name: /CollectionInfo/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
