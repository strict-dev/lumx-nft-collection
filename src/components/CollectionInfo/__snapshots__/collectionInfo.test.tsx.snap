// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollectionInfo /> should render the heading 1`] = `
<div
  class="mx-4 pt-20 "
>
  <div
    class="p-3 rounded-sm bg-neutral-800 mb-14"
  >
    <div
      class="relative flex items-center justify-between gap-2 pb-4"
    >
      <div
        class="flex items-center justify-between gap-2"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2732%27%20height=%2732%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="User"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="flex flex-col justify-between"
        >
          <h1
            class="text-base font-bold"
          >
            Bored Ape Yacht Club
          </h1>
          <div
            class="flex"
          >
            <p
              class="text-sm font-normal truncate max-w-[140px] text-neutral-600"
            >
              0x7b123f53421b1bF8533339BFBdc7C98aA94163db
            </p>
            <button>
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="User"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="flex gap-2 "
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2732%27%20height=%2732%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Monitor Icon"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2732%27%20height=%2732%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="OpenSea Icon"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2732%27%20height=%2732%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="EtherScan Icon"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
    <hr
      class="border-neutral-700"
    />
    <div
      class="grid grid-flow-row grid-cols-3 grid-rows-2 gap-2 pt-4"
    >
      <div
        class="flex flex-col items-center justify-center gap-1 box"
      >
        <p
          class="text-xs font-bold text-neutral-600"
        >
          LIFETIME
        </p>
        <div
          class="flex items-center gap-1"
        >
          <span
            class="text-lg font-bold"
          >
            100d
          </span>
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center gap-1 box"
      >
        <p
          class="text-xs font-bold text-neutral-600"
        >
          ITEMS
        </p>
        <div
          class="flex items-center gap-1"
        >
          <span
            class="text-lg font-bold"
          >
            10k
          </span>
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center gap-1 box"
      >
        <p
          class="text-xs font-bold text-neutral-600"
        >
          FLOOR
        </p>
        <div
          class="flex items-center gap-1"
        >
          <span
            class="text-lg font-bold"
          >
            145
          </span>
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center gap-1 box"
      >
        <p
          class="text-xs font-bold text-neutral-600"
        >
          ROYALTIES
        </p>
        <div
          class="flex items-center gap-1"
        >
          <span
            class="text-lg font-bold"
          >
            7.5%
          </span>
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center gap-1 box"
      >
        <p
          class="text-xs font-bold text-neutral-600"
        >
          VOL24H
        </p>
        <div
          class="flex items-center gap-1"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <span
            class="text-lg font-bold"
          >
            145
          </span>
        </div>
        <div
          class="flex items-center justify-center gap-1"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <span
            class="font-bold text-support-success"
          >
            12.02%
          </span>
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center gap-1 box"
      >
        <p
          class="text-xs font-bold text-neutral-600"
        >
          VOL7D
        </p>
        <div
          class="flex items-center gap-1"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <span
            class="text-lg font-bold"
          >
            145
          </span>
        </div>
        <div
          class="flex items-center justify-center gap-1"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <span
            class="font-bold text-support-error"
          >
            1.29%
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
